share/json.py

```diff
@@ -0,0 +1,15 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+
+from typing import Any, AnyStr
+
+import ujson
+
+
+def json_dumper(json_object: Any) -> str:
```

Comment:
that will be hard to maintain when we want to mock for benchmark etc, I suspect

also different libraries have different arguments

in the case we refactor like you propose will have to mock change the implementation of `WrappedJson`?
but this will require in the end writing N `WrappedJson*` like we have N `json_dumper*`, did I get it right?

what will make the difference is using `from shared import json` instead of `import json`, but this indeed for me it's not explicit enough about the fact that we are indeed using a custom json lib and it's error prone since I see we could easily forget to import json from `shared` instead that from the stdlib

since we are not providing a package to be consumed from third party developers, having a more explicit `json_dumper` make me more comfortable on this

not sure, we can discuss further and add the refactoring later on

Reply:
we have the issue of forgetting to use the shared json lib regardless of how we name/implement the wrappers. one mitigation for this is instrumenting the import of stdlib `json` at test time and throwing an exception. another is adding some additional comments into the shared json lib that explains the importance of ujson vs stdlib for proper functioning of the forwarder.

the mocking piece seems fairly trivial either way - in this case we could inject the base lib into the constructor, and for any different arguments just overwrite the affected members. 

all that said i really don't mind how we do it. my main thing was that we are introducing really non-standard naming. in particular the `json_dumper/loader` names reads to me like they're factories. but it's not worth spending too much time on. i'm happy with this as is, so we can leave it.

Reply:
> we have the issue of forgetting to use the shared json lib regardless of how we name/implement the wrappers. one mitigation for this is instrumenting the import of stdlib `json` at test time (or even runtime) and throwing an exception. another is adding some additional comments into the shared json lib that explains the importance of ujson vs stdlib for proper functioning of the forwarder.

what I mean is that looking through the code having `json_dumper`/`json_loader` it will be a more evident hint that we are not using the stdlib `json` or we are wrapping it at some degree.
also I have a current branch testing [hyperscan](https://github.com/intel/hyperscan) for the multiline and include/exclude filter, and while the API is different from stdlib `re`, even if they would be the same, there is a case for the AWS secret manager expander, where I don't want to replace them. this lead me think that similarly there could be some case where we will be fine with stdlib `json` and in this case it would just be a too much subtle for me `import json` vs `from share import json` to make the difference :)

> all that said i really don't mind how we do it. my main thing was that we are introducing really non-standard naming. in particular the `json_dumper/loader` names reads to me like they're factories. but it's not worth spending too much time on. i'm happy with this as is, so we can leave it.

as you can see my main concern regarding the comment was about readability :)
In this regard the fact that we use non-standard naming is somehow intended, as said to hint about using non-standard module.
at the same time if `json_dumper/loader` names read like they're factories, I'm totally open to find better names :)

No need to block merging for this, unless you have a prompt alternatives :), but let's improve the naming in the future